cupy/cuda/compiler.py

```diff
@@ -475,6 +485,11 @@ def compile(self, options=()):
                 mapping = {}
                 for ker in self.name_expressions:
                     mapping[ker] = nvrtc.getLoweredName(self.ptr, ker)
+            if log_stream == 'stdout':
+                print(nvrtc.getProgramLog(self.ptr))
```

Comment:
use the stream directly instead of calling print.

Reply:
Same here?

```python
sys.stdout = nvrtc.getProgramLog(self.ptr)